pkg/apis/core/types.go

```diff
@@ -2235,9 +2229,8 @@ type Lifecycle struct {
 	// PreStop is called immediately before a container is terminated due to an
 	// API request or management event such as liveness/startup probe failure,
 	// preemption, resource contention, etc. The handler is not called if the
-	// container crashes or exits. The reason for termination is passed to the
-	// handler. The Pod's termination grace period countdown begins before the
-	// PreStop hooked is executed. Regardless of the outcome of the handler, the
+	// container crashes or exits. The Pod's termination grace period countdown begins before the
+	// PreStop hook is executed. Regardless of the outcome of the handler, the
 	// container will eventually terminate within the Pod's termination grace
 	// period. Other management of the container blocks until the hook completes
```

Comment:
Add a couple words on finalizers here?

Reply:
what do you have in mind? Just reference them?

Reply:
Something like "will eventually terminate within the Pod's termination grace period (unless delayed by finalizers). " ?

Reply:
thank you!